src/pip/_internal/build_env.py

```diff
@@ -74,12 +72,37 @@ def restore_var(varname, old_value):
     def cleanup(self):
         self._temp_dir.cleanup()
 
+    def install_requirements(self, finder, requirements, message):
+        args = [
+            sys.executable, '-m', 'pip', 'install', '--ignore-installed',
```

Comment:
What's the reason for adding `--ignore-installed` here?

Reply:
This call is not run with build isolation.

Reply:
Aha.

Isolating this run would make pip not visible -> `python -m pip` won't work, and a temporary prefix for the installation of packages makes this non env-breaking.